src/main/java/com/google/sps/data/Post.java

```diff
@@ -1,21 +1,35 @@
 package com.google.sps.data;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import com.google.sps.data.Comment;
+import com.google.sps.data.Reaction;
 
 public final class Post{
     private final Comment question;
     private final Comment answer;
     private final List<Comment> replies;
     private final long id; 
+    private final Map<Reaction, Long> reactions; 
 
-    public Post(Comment question, Comment answer, List<Comment> replies, long id){
+    public Post(Comment question, Comment answer, List<Comment> replies, long id, Map<Reaction, Long> reactions){
         this.question = question;
         this.answer = answer;
         this.replies = replies;
         this.id = id;
+        this.reactions = reactions; 
+    }
+
+    public Map<Reaction, Long> getReactions() {
+        Map<Reaction, Long> copy = new HashMap<Reaction, Long>();
+        for (Map.Entry<Reaction, Long> entry : reactions.entrySet())
```

Comment:
I believe Hashmap has a constructor that can take another map - see if that helps.